geo/src/algorithm/bool_ops/i_overlay_integration.rs

```diff
@@ -0,0 +1,323 @@
+use crate::geometry::Coord;
+use crate::GeoNum;
+use i_overlay::core::fill_rule::FillRule;
+use i_overlay::core::overlay::ShapeType;
+use i_overlay::core::overlay_rule::OverlayRule;
+use i_overlay::string::clip::ClipRule;
+
+pub trait BoolOpsCoord<T>: Copy {
+    fn new(x: T, y: T) -> Self;
+    fn x(&self) -> T;
+    fn y(&self) -> T;
+}
+
+/// A geometry coordinate number suitable for performing geometric boolean operations.
+pub trait BoolOpsNum: GeoNum {
+    type CoordType: BoolOpsCoord<Self>;
+    type OverlayType: BoolOpsOverlay<CoordType = Self::CoordType>;
+    type StringOverlayType: BoolOpsStringOverlay<CoordType = Self::CoordType>;
+
+    fn to_bops_coord(geo_coord: Coord<Self>) -> Self::CoordType {
+        Self::CoordType::new(geo_coord.x, geo_coord.y)
+    }
+
+    fn to_geo_coord(bops_coord: Self::CoordType) -> Coord<Self> {
+        Coord {
+            x: bops_coord.x(),
+            y: bops_coord.y(),
+        }
+    }
+}
+
+pub trait BoolOpsOverlay {
+    type CoordType;
+    type OverlayGraph: BoolOpsOverlayGraph<CoordType = Self::CoordType>;
+    fn new() -> Self;
+    fn add_path(&mut self, path: Vec<Self::CoordType>, shape_type: ShapeType);
+    fn into_graph(self, fill_rule: FillRule) -> Self::OverlayGraph;
+}
+
+pub(super) trait BoolOpsOverlayGraph {
+    type CoordType;
+    fn extract_shapes(&self, overlay_rule: OverlayRule) -> Vec<Vec<Vec<Self::CoordType>>>;
+}
+
+pub trait BoolOpsStringOverlay {
+    type CoordType;
+    type StringGraph: BoolOpsStringGraph<CoordType = Self::CoordType>;
+    fn new() -> Self;
+    fn add_shape_path(&mut self, path: Vec<Self::CoordType>);
+    fn add_string_line(&mut self, path: [Self::CoordType; 2]);
+    fn into_graph(self, fill_rule: FillRule) -> Self::StringGraph;
+}
+
+pub(super) trait BoolOpsStringGraph {
+    type CoordType;
+    fn clip_string_lines(&self, clip_rule: ClipRule) -> Vec<Vec<Self::CoordType>>;
+}
+
+mod f64 {
+    use super::{ClipRule, FillRule, OverlayRule, ShapeType};
+    use i_overlay::f64::{
+        graph::F64OverlayGraph,
+        overlay::F64Overlay,
+        string::{F64StringGraph, F64StringOverlay},
+    };
+    use i_overlay::i_float::f64_point::F64Point;
```

Comment:
I wonder if they'd be open to incorporating `num-traits` in the future. Might simplify some of this?

Reply:
Very likely they just didnt know about it, the maintainer seems very nice so you could probably just ask them over at their github

https://github.com/iShape-Rust/iOverlay/issues